clamav/cnudie.py

```diff
@@ -21,22 +25,103 @@
 ci.log.configure_default_logging()
 
 
+def _should_skip_scan(
+        resource_node: cnudie.iter.ResourceNode,
+        current_version_info: clamav.model.ClamAVVersionInfo,
+        virus_db_max_age_days: int,
+        skip_if_virus_definition_unchanged: bool,
+        delivery_client: delivery.client.DeliveryServiceClient = None,
+) -> bool:
+
+    if not delivery_client:
+        return False
+
+    previous_findings = delivery_client.artefact_metadata_for_resource_node(
+        resource_node=resource_node,
+        types=[dso.model.Datatype.MALWARE],
+    )
+
+    sorted_findings = sorted(
+        previous_findings,
+        key=lambda f: f.meta.creation_date.astimezone(),
+        reverse=True,
+    )
+
+    if sorted_findings:
```

Comment:
codestyle: rather place early-exit first, add non-early-exit path outside of if-block